spec/std/spec/expectations_spec.cr

```diff
@@ -23,6 +23,18 @@ describe "expectations" do
     it { 100_000.should_not be_a(String) }
     it { 100_000.should be_a(Int32) }
     it { "Hello".should_not be_a(Int32) }
+
+    it "restricts type on should" do
+      x = 1 || 'a'
+      y = x.should be_a(Int32)
+      typeof(y).should eq(Int32)
```

Comment:
Perhaps verify `typeof(x).should eq(Int32 | Char)` in case the type checker gets cleverer and the spec isnt updated.

Reply:
Done!